src/app/chat/view.nim

```diff
@@ -75,6 +76,24 @@ QtObject:
   proc getChannelColor*(self: ChatsView, channel: string): string {.slot.} =
     self.chats.getChannelColor(channel)
 
+  proc replyAreaEnabled*(self: ChatsView, enable: bool, userName: string, message: string, identicon: string) {.signal.}
+
+  proc setReplyTo(self: ChatsView, messageId: string) {.slot.} =
+    self.replyTo = messageId
+
+  proc enableReplyArea*(self: ChatsView, enable: bool, userName: string = "", message: string = "", identicon: string = "") {.slot.} =
+    if not enable:
+      self.replyTo = ""
+    self.replyAreaEnabled(enable, username, message, identicon)
+  
+  proc disableReplyArea(self: ChatsView) =
+    self.replyTo = ""
+    self.replyAreaEnabled(false, "", "", "")
```

Comment:
All the code above for the replyArea should be possible to be handled in QML IMO. This is mostly just local state handling that the QML would be better tooled to handle.